contracts/src/Assets.sol

```diff
@@ -144,7 +148,18 @@ library Assets {
         AssetsStorage.Layout storage $ = AssetsStorage.layout();
 
         // Lock the funds into AssetHub's agent contract
-        _transferToAgent($.assetHubAgent, token, sender, amount);
+        if (token != address(0)) {
```

Comment:
I think the way you've introduced native ether support adds complexity by adding a lot of decision branches at the lowest level.

We already have branches at a higher level to choose between foreign, native tokens. We can add a branch for eth support there.

Concretely, I would suggest adding a new `_sendEther` function instead of adding branches in `_sendNativeToken`.

Reply:
Addressed in [d8a9a58](https://github.com/Snowfork/snowbridge/pull/1354/commits/d8a9a58d6572f619ef00c8a57f44c2022ffa9b56)

Reply:
It looks like you reverted the `_sendEther` and `_sendNativeToken` split and merged it again. Just out of interest why?

Reply:
changed my mind, your original way was cleaner :)